src/main/java/org/folio/rest/impl/PrecedingSucceedingTitleAPI.java

```diff
@@ -68,6 +94,56 @@ public void putPrecedingSucceedingTitlesByPrecedingSucceedingTitleId(String prec
       PutPrecedingSucceedingTitlesByPrecedingSucceedingTitleIdResponse.class, asyncResultHandler);
   }
 
+  private boolean validatePrecedingSucceedingTitles(List<PrecedingSucceedingTitle> precedingSucceedingTitles,
```

Comment:
Maybe it will be better to move this logic and  saveCollection into service layer?

Reply:
It's is a good point. But in this case, we should move all logic of this interface implementation to the service layer, not only one endpoint. I think there could be a separate story for this if it is needed.